packages/backend-indexer/src/actions/indexer/hyperlaneWarpRoute.ts

```diff
@@ -32,48 +36,52 @@ export async function getHyperlaneRoutes<chain extends Chain | undefined>(
         console.warn(`Failed to fetch wrapped token for router ${address}:`, error);
     }
 
-    const domains: number[] = [
-        ...(await readContract(client, {
-            address,
-            abi: Router.abi,
-            functionName: "domains",
-        })),
-    ];
+    const domains = await readContract(client, {
+        address,
+        abi: Router.abi,
+        functionName: "domains",
+    });
 
-    const promises = domains.map((domain) =>
-        readContract(client, {
-            address,
-            abi: Router.abi,
-            functionName: "routers",
-            args: [domain],
-        })
-            .then((routerAddress: Hex) => (routerAddress !== zeroAddress ? { domain, router: routerAddress } : null))
-            .catch((error) => {
-                console.error(`Failed to fetch router for domain ${domain}:`, error);
-                return null;
-            }),
+    const invalidDomains = domains.filter((domain) => !chainIds.slice(1).includes(domain));
+    if (invalidDomains.length > 0) {
+        throw new Error(`Invalid domains found: ${invalidDomains.join(", ")}.`);
+    }
+
+    const routerResults = await Promise.all(
+        domains.map((domain) =>
+            readContract(client, {
+                address,
+                abi: Router.abi,
+                functionName: "routers",
+                args: [domain],
+            })
+                .then((routerAddress) => (routerAddress !== zeroAddress ? { domain, router: routerAddress } : null))
+                .catch((error) => {
+                    console.error(`Failed to fetch router for domain ${domain}:`, error);
+                    return null;
+                }),
+        ),
     );
 
-    const results = await Promise.all(promises);
```

Comment:
Nothing wrong with keeping the results on another line.
Just call it routerPromises instead of just promises though